### ISSUE TITLE
Dev branch of tm-sme/Lib_UNO crashes GraphDB

### ISSUE THREAD

**Issue by romichg** [TeamMentor/TM_4_0_Design#902]
to duplicate:
cd data/Lib_UNO
git checkout Dev
git pull

cd ../../code/TM_GraphDB
node app.js

Hit the index

```
$ node app.js
[SiteData] loading data from c:\TEAM Mentor-DEV\TMDEV01-Site_beta.teammentor.net\TM40\Tm\config\SiteData
[SiteData] Custom code loaded: express_Routes
Hooking Console to Winston logger
[Logging-Service] console hooked
2015-07-25T08:18:00.809Z - info: Adding swagger support
2015-07-25T08:18:01.496Z - info: Server started at: http://localhost:10002
2015-07-25T08:18:06.043Z - info:  method=GET, url=/data/library_Query
2015-07-25T08:18:06.059Z - info:  method=GET, url=/convert/to_ids/query-6234f2d47eb7
2015-07-25T08:18:06.309Z - info:  method=GET, url=/data/query_tree/query-6234f2d47eb7
2015-07-25T08:18:07.074Z - info: [get_Queries_Mappings] Fixed parent articles for query query-6234f2d47eb7
2015-07-25T08:18:07.074Z - info: [get_Queries_Mappings] Fixed parent articles for query query-744ce901584b
c:\TEAM Mentor-DEV\TMDEV01-Site_beta.teammentor.net\TM40\Tm\code\TM_GraphDB\src\services\data\Query-Tree.coffee:238
          results.push(container.title.lower());
                                       ^
TypeError: undefined is not a function
  at c:\TEAM Mentor-DEV\TMDEV01-Site_beta.teammentor.net\TM40\Tm\code\TM_GraphDB\src\services\data\Query-Tree.coffee:152:31
  at Query_Tree.sort_Containers (c:\TEAM Mentor-DEV\TMDEV01-Site_beta.teammentor.net\TM40\Tm\code\TM_GraphDB\src\services\data\Que
  at c:\TEAM Mentor-DEV\TMDEV01-Site_beta.teammentor.net\TM40\Tm\code\TM_GraphDB\src\services\data\Query-Tree.coffee:43:37
  at c:\TEAM Mentor-DEV\TMDEV01-Site_beta.teammentor.net\TM40\Tm\code\TM_GraphDB\src\services\data\Query-Mappings.coffee:104:7
```

**Comment by DinisCruz** [comment]
I can't duplicate this on the latest commits you pushed into Lib_UNO and Lib_UNO-json

The root cause of the issue is the fact that there was a query without a title

I started looking at it, and the fix needs to happen a bit upstream from the location where the code is (although I will still improve it so that if the code reaches that location it doesn't fail like it does now)

**Comment by romichg** [comment]
This will not happen on latest Lib_UNO.  As I noted, this will happen when you checkout the Dev branch of Lib_UNO

**Comment by DinisCruz** [comment]
does this still happen?

**Comment by romichg** [comment]
yes, please fix.

```
2015-09-08T14:56:20.565Z - info:  url=/show/, ip=::1, agent=undefined
2015-09-08T14:56:30.670Z - info:  url=/jade/show/, ip=::1, agent=undefined
2015-09-08T14:56:30.681Z - info:  method=GET, url=/data/library_Query
2015-09-08T14:56:30.700Z - info:  user=rTest5, action=show, queryId=query-6234f2d47eb7, filters=undefined
2015-09-08T14:56:30.706Z - info:  method=GET, url=/convert/to_ids/query-6234f2d47eb7
2015-09-08T14:56:30.712Z - info: ***[Graph-Service]*** open db
2015-09-08T14:56:30.734Z - info: ***[Graph-Service]*** close db
2015-09-08T14:56:30.740Z - info:  method=GET, url=/data/query_tree/query-6234f2d47eb7
2015-09-08T14:56:30.948Z - info:  Error=There are no results that match the search., queryId=query-6234f2d47eb7, filters=undefined
TypeError: undefined is not a function
  at /home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:180:27
  at Search_Query_Tree.sort_Queries (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:302:9)
  at Search_Query_Tree.map_Queries_For_Query (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:87:14)
  at Search_Query_Tree.map_Queries_For_Query (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:1:1)
  at Search_Query_Tree.create_Query_Tree_For_Query_Id (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:30:22)
  at Search_Query_Tree.create_Query_Tree_For_Query_Id (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:1:1)
  at Query_Tree.create_Query_Tree (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/query-tree/Query-Tree.coffee:28:27)
  at Query_Tree.create_Query_Tree (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/query-tree/Query-Tree.coffee:1:1)
  at Query_Tree.get_Query_Tree_Filtered (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/query-tree/Query-Tree.coffee:57:11)
  at Query_Tree.get_Query_Tree_Filtered (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/query-tree/Query-Tree.coffee:1:1)
  at Query_Tree.get_Query_Tree (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/query-tree/Query-Tree.coffee:42:7)
  at Query_Tree.get_Query_Tree (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/query-tree/Query-Tree.coffee:1:1)
  at Data_API.query_tree (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/api/Data-API.coffee:107:22)
  at Data_API.query_tree (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/api/Data-API.coffee:1:1)
  at get_Command.action (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/api/base-classes/Swagger-Common.coffee:19:36)
```

**Comment by romichg** [comment]
It is importaint we figure this one out. @DinisCruz I am counting on your help for this one. Although it is not extremely urgent.  Putting this in Spring 2. (we are back to sprint 1 now)

**Comment by DinisCruz** [comment]
error is happening here

```
/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:180:27 
```

![image](https://cloud.githubusercontent.com/assets/656739/10317260/74754ae0-6c5b-11e5-88c9-1bd3ed270d66.png)

which happens when there is a query with no title

**Comment by romichg** [comment]
I don't think we have a view/folder with no title in the data..

**Comment by DinisCruz** [comment]
@roman87 can you try with the fix pushed?

**Comment by romichg** [comment]
Nope, still an issue:

```
rom@TITI09:~/Dev/TeamMentor/TM$ git submodule
 4cc3a0c002fad180f86d32ccab22fc74998c675e build/TM_4_0_JSON_Data (4cc3a0c)
 4321dcc3823fd3abc0e5b910a1d8118ac61a215c build/TM_Gulp (heads/master)
 80a82a640cf9de399e9150e58dfdce3a338f3557 code/TM_Angular (heads/Dev)
 4c28aa0325918af9f19345865974deefce246f1c code/TM_Flare (4.0.3-336-g4c28aa0)
+d660ef71075c44a875ce9c631dd5937548d2bbbd code/TM_GraphDB (4.0.1-37-gd660ef7)
 8bf0b3c7da2a7c5e654a8ead211e36d124b4d974 code/TM_Jade (heads/Dev)
 c8406320efbbdb8986823fce6935499c0f7ae470 code/TM_Shared (heads/Dev)
 7ab139c0b438a4cae7312346cfdce2b2c823e55e code/TM_Static (heads/Dev)
 c4777a5e078566c5bcf58c7c7d2858c6e24d9234 code/TM_Website (4.0.3-152-gc4777a5)
 6be8ca14391ba9582bf42f691a5e48dc8f11663d config/SiteData_TM (heads/Dev)
 dd65a11bf320940d515c8e06870616a5ae8a03f1 data/Lib_Docs-json (heads/master)
+c8baba356996cf1b6e3ac03a85e2f7083128f981 data/Lib_UNO (heads/Dev)
 95e9f6fc9d672de1d36b4bf3d5026a12158d362a data/Lib_UNO-json (heads/Beta)
 921ea5e009deca34bc3b34fbd7e54ec746f0fa87 qa/TM_QA (4.0.1-34-g921ea5e)
```

```
[Logging-Service] console hooked
2015-10-15T13:57:55.676Z - info: [TM-Server] Log is setup
2015-10-15T13:57:55.697Z - info: Configuring TM_Design Express server
Hooking Console to Winston logger
[Logging-Service] console hooked
2015-10-15T13:57:57.680Z - info: Adding swagger support
2015-10-15T13:58:00.033Z - info: Server started at: http://localhost:12346
2015-10-15T13:58:00.042Z - info: ***[Graph-Service]*** open db
2015-10-15T13:58:00.043Z - info: [ensure_TM_Uno_Is_Loaded] Using path_To_Lib_Uno_Json: ../../data/Lib_UNO-json
2015-10-15T13:58:00.044Z - info: [ensure_TM_Uno_Is_Loaded] tm-uno-loaded.flag file doesn't exist, so deleting GraphDB and re-importing Lib_Uno-Json data
2015-10-15T13:58:00.044Z - info: ***[Graph-Service]*** close db
2015-10-15T13:58:00.046Z - info: [ensure_TM_Uno_Is_Loaded] deleting data_cache and search_cache folders
2015-10-15T13:58:00.064Z - info: [ensure_TM_Uno_Is_Loaded] Loading graph data from: lib-uno-triplets.json
2015-10-15T13:58:00.407Z - info: [ensure_TM_Uno_Is_Loaded] There are 35181 triplets to import
2015-10-15T13:58:01.081Z - info: Starting 'TM Jade' Poc on port 12345
2015-10-15T13:58:01.086Z - info: Analytics not enabled
2015-10-15T13:58:01.092Z - info: Hubspot not enabled
2015-10-15T13:58:01.111Z - info: Cleaning expired records from _anonymousVisits datastore...
2015-10-15T13:58:01.161Z - info: Number of expired _anonymousVisits records removed: 0
2015-10-15T13:58:01.172Z - info: [Session_Service] clearing empty sessions
2015-10-15T13:58:01.174Z - info: [Session_Service] Configured
2015-10-15T13:58:02.388Z - info: [ensure_TM_Uno_Is_Loaded] Loading graph data from: virtual-articles.json
2015-10-15T13:58:02.389Z - info: [ensure_TM_Uno_Is_Loaded] There are 29 triplets to import
2015-10-15T13:58:07.317Z - info: [ensure_TM_Uno_Is_Loaded] Data loaded at Thu Oct 15 2015 09:58:07 GMT-0400 (EDT)
2015-10-15T13:58:07.330Z - info: %s: %dms
2015-10-15T13:58:07.330Z - info: ***[Graph-Service]*** close db
2015-10-15T13:58:12.432Z - info: ***[Graph-Service]*** close db
2015-10-15T13:58:12.540Z - info: ***[Graph-Service]*** open db
2015-10-15T13:58:12.603Z - info: ***[Graph-Service]*** close db
2015-10-15T13:58:12.609Z - info: ***[Graph-Service]*** open db
2015-10-15T13:58:14.086Z - info: [get_Queries_Mappings] Fixed parent articles for query query-6234f2d47eb7
2015-10-15T13:58:14.092Z - info: [get_Queries_Mappings] Fixed parent articles for query query-744ce901584b
2015-10-15T13:58:14.110Z - info: ***[Graph-Service]*** close db
2015-10-15T13:58:14.196Z - info: ***[Graph-Service]*** open db
2015-10-15T13:58:14.202Z - info: ***[Graph-Service]*** close db
2015-10-15T13:58:14.206Z - info: ***[Graph-Service]*** open db
2015-10-15T13:58:14.242Z - info: ***[Graph-Service]*** close db
2015-10-15T13:58:14.286Z - info: ***[Graph-Service]*** open db
2015-10-15T13:58:14.514Z - info: ***[Graph-Service]*** close db
2015-10-15T13:58:14.518Z - info: ***[Graph-Service]*** open db
2015-10-15T13:58:14.532Z - info: 0
2015-10-15T13:58:14.535Z - info: ***[Graph-Service]*** close db
2015-10-15T13:58:14.537Z - info: ***[Graph-Service]*** open db
2015-10-15T13:58:14.622Z - info: [exec_In_DB][async error]
2015-10-15T13:58:14.623Z - info: ***[Graph-Service]*** close db
2015-10-15T13:58:14.625Z - info: [TM-Server][search_Setup.build_All] completed
2015-10-15T13:58:26.874Z - info:  url=/, ip=::1, agent=undefined
2015-10-15T13:58:26.903Z - info:  url=/angular/user/index, ip=::1, agent=undefined
2015-10-15T13:58:26.924Z - info:  url=/angular/guest/home, ip=::1, agent=undefined
2015-10-15T13:58:28.139Z - info:  url=/angular/js/lib.js, ip=::1, agent=undefined
2015-10-15T13:58:28.151Z - info:  url=/angular/js/code.js, ip=::1, agent=undefined
2015-10-15T13:58:28.612Z - info:  url=/angular/jade-html/component/navigation/landing_bar, ip=::1, agent=undefined
2015-10-15T13:58:28.764Z - info:  url=/angular/jade-html/views/guest/home, ip=::1, agent=undefined
2015-10-15T13:58:29.099Z - info: Transformers.md is deprecated, you must replace the :md jade filter, with :marked and install jstransformer-marked before you update to jade@2.0.0.
2015-10-15T13:58:31.136Z - info:  url=/angular/jade-html/views/guest/login, ip=::1, agent=undefined
2015-10-15T13:58:31.688Z - info:  url=/angular/jade-html/component/guest/login_form, ip=::1, agent=undefined
2015-10-15T13:58:39.639Z - info:  url=/json/user/login, ip=::1, agent=undefined
2015-10-15T13:58:45.445Z - info:  url=/json/user/currentuser, ip=::1, agent=undefined
2015-10-15T13:58:45.644Z - info:  url=/jade/json/tm/config, ip=::1, agent=undefined
2015-10-15T13:58:45.665Z - info:  url=/angular/user/index, ip=::1, agent=undefined
2015-10-15T13:58:46.113Z - info:  url=/angular/user/index, ip=::1, agent=undefined
2015-10-15T13:58:46.218Z - info:  url=/angular/js/lib.js, ip=::1, agent=undefined
2015-10-15T13:58:46.223Z - info:  url=/angular/js/code.js, ip=::1, agent=undefined
2015-10-15T13:58:46.621Z - info:  url=/angular/jade-html/component/user/search_bar, ip=::1, agent=undefined
2015-10-15T13:58:46.687Z - info:  url=/angular/jade-html/component/navigation/left_navigation, ip=::1, agent=undefined
2015-10-15T13:58:46.768Z - info:  url=/angular/jade-html/views/user/index, ip=::1, agent=undefined
2015-10-15T13:58:46.891Z - info:  url=/api/data/query_view_model/query-6234f2d47eb7/0/0, ip=::1, agent=undefined
2015-10-15T13:58:46.969Z - info:  method=GET, url=/data/query_view_model/query-6234f2d47eb7/0/0
2015-10-15T13:58:47.056Z - info:  url=/angular/jade-html/component/user/queries, ip=::1, agent=undefined
2015-10-15T13:58:47.199Z - info:  url=/angular/jade-html/component/user/breadcrumbs, ip=::1, agent=undefined
2015-10-15T13:58:47.239Z - info:  url=/angular/jade-html/component/user/filters_active, ip=::1, agent=undefined
2015-10-15T13:58:47.308Z - info:  url=/angular/jade-html/component/user/results, ip=::1, agent=undefined
2015-10-15T13:58:47.338Z - info:  url=/angular/jade-html/component/user/articles, ip=::1, agent=undefined
2015-10-15T13:58:47.385Z - info:  url=/angular/jade-html/component/user/pagination, ip=::1, agent=undefined
2015-10-15T13:58:47.542Z - info:  url=/angular/jade-html/component/user/found_issue, ip=::1, agent=undefined
2015-10-15T13:58:47.668Z - info:  url=/angular/jade-html/component/user/filters, ip=::1, agent=undefined
2015-10-15T13:58:47.782Z - info:  url=/json/user/currentuser, ip=::1, agent=undefined
2015-10-15T13:58:47.786Z - info:  url=/angular/jade-html/component/user/article_box, ip=::1, agent=undefined
2015-10-15T13:58:47.889Z - info:  url=/jade/json/tm/config, ip=::1, agent=undefined
2015-10-15T13:58:47.915Z - info:  url=/api/data/query_view_model/query-6234f2d47eb7/0/10, ip=::1, agent=undefined
TypeError: undefined is not a function
  at Search_Query_Tree.sort_Queries (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:185:24)
  at Search_Query_Tree.map_Queries_For_Query (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:87:14)
  at Search_Query_Tree.map_Queries_For_Query (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:1:1)
  at Search_Query_Tree.create_Query_Tree_For_Query_Id (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:30:22)
  at Search_Query_Tree.create_Query_Tree_For_Query_Id (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:1:1)
  at Query_Tree.create_Query_Tree (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/query-tree/Query-Tree.coffee:28:27)
  at Query_Tree.create_Query_Tree (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/query-tree/Query-Tree.coffee:1:1)
  at Query_Tree.get_Query_Tree_Filtered (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/query-tree/Query-Tree.coffee:57:11)
  at Query_Tree.get_Query_Tree_Filtered (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/query-tree/Query-Tree.coffee:1:1)
  at Query_View_Model.query_Tree_Filtered (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/data/Query-View-Model.coffee:78:18)
  at Query_View_Model.query_Tree_Filtered (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/data/Query-View-Model.coffee:1:1)
  at Query_View_Model.get_View_Model (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/data/Query-View-Model.coffee:51:7)
  at Query_View_Model.get_View_Model (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/data/Query-View-Model.coffee:1:1)
  at Data_API.query_view_model (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/api/Data-API.coffee:132:28)
  at Data_API.query_view_model (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/api/Data-API.coffee:1:1)
  at get_Command.action (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/api/base-classes/Swagger-Common.coffee:19:36)
  at /home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/swagger-node-express/lib/swagger.js:424:9
  at Layer.handle [as handle_request] (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/layer.js:95:5)
  at next (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/route.js:131:13)
  at Route.dispatch (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/route.js:112:3)
  at Layer.handle [as handle_request] (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/layer.js:95:5)
  at /home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:277:22
  at param (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:349:14)
  at param (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:365:14)
  at param (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:365:14)
  at param (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:365:14)
  at Function.process_params (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:410:3)
  at next (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:271:10)
  at jsonParser (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/body-parser/lib/types/json.js:100:40)
  at Layer.handle [as handle_request] (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/layer.js:95:5)
  at trim_prefix (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:312:13)
  at /home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:280:7
  at Function.process_params (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:330:12)
  at next (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:271:10)
  at urlencodedParser (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/body-parser/lib/types/urlencoded.js:88:40)
  at Layer.handle [as handle_request] (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/layer.js:95:5)
  at trim_prefix (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:312:13)
  at /home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:280:7
  at Function.process_params (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:330:12)
  at next (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:271:10)
  at /home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/TM-Server.coffee:32:9
  at Layer.handle [as handle_request] (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/layer.js:95:5)
  at trim_prefix (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:312:13)
  at /home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:280:7
  at Function.process_params (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:330:12)
  at next (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:271:10)
  at compression (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/compression/index.js:218:5)
  at Layer.handle [as handle_request] (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/layer.js:95:5)
  at trim_prefix (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:312:13)
  at /home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:280:7
  at Function.process_params (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:330:12)
  at next (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:271:10)
  at expressInit (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/middleware/init.js:33:5)
  at Layer.handle [as handle_request] (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/layer.js:95:5)
  at trim_prefix (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:312:13)
  at /home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:280:7
  at Function.process_params (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:330:12)
  at next (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:271:10)
  at query (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/middleware/query.js:49:5)
  at Layer.handle [as handle_request] (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/layer.js:95:5)
  at trim_prefix (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:312:13)
  at /home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:280:7
  at Function.process_params (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:330:12)
  at next (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:271:10)
  at Function.handle (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/router/index.js:176:3)
  at EventEmitter.handle (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/application.js:173:10)
  at Server.app (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/node_modules/express/lib/express.js:38:9)
  at Server.emit (events.js:110:17)
  at HTTPParser.parserOnIncoming [as onIncoming] (_http_server.js:491:12)
  at HTTPParser.parserOnHeadersComplete (_http_common.js:111:23)
  at Socket.socketOnData (_http_server.js:343:22)
  at Socket.emit (events.js:107:17)
  at readableAddChunk (_stream_readable.js:163:16)
  at Socket.Readable.push (_stream_readable.js:126:10)
  at TCP.onread (net.js:538:20)

2015-10-15T13:58:48.354Z - info:  method=GET, url=/data/query_view_model/query-6234f2d47eb7/0/10
TypeError: undefined is not a function
  at Search_Query_Tree.sort_Queries (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:185:24)
  at Search_Query_Tree.map_Queries_For_Query (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:87:14)
  at Search_Query_Tree.map_Queries_For_Query (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:1:1)
  at Search_Query_Tree.create_Query_Tree_For_Query_Id (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:30:22)
  at Search_Query_Tree.create_Query_Tree_For_Query_Id (/home/rom/Dev/TeamMentor/TM/code/TM_GraphDB/src/services/search/Search-Query-Tree.coffee:1:1)
```

**Comment by romichg** [comment]
No longer the case
